.github/workflows/docs.yml

```diff
@@ -13,32 +13,40 @@
 
 name: Documentation
 on:
-  pull_request:
-    types: [opened, reopened, synchronize]
+  pull_request_target:
+    types: [opened, reopened, synchronize] # Handles forked PRs
   push:
   merge_group:
     types: [checks_requested]
+
 jobs:
   docs-build:
     name: Build documentation
     runs-on: ubuntu-latest
     permissions:
       pull-requests: write
     steps:
-      - name: Checkout repository
+      - name: Checkout repository (Handle all events)
         uses: actions/checkout@v4.2.2
+        with:
+          ref: ${{ github.head_ref || github.event.pull_request.head.ref || github.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name || github.repository }}
+
       - name: Setup Bazel
         uses: bazel-contrib/setup-bazel@0.9.1
+
       - name: Build documentation
         run: |
           bazel build //docs:github-pages && cp bazel-bin/docs/github-pages.tar .
+
       - name: Upload artifact for job analysis
         uses: actions/upload-artifact@v4.4.0
         with:
-          name: github-pages-${{ github.sha }}
+          name: github-pages-${{ github.event.pull_request.head.sha || github.sha }}
```

Comment:
Same here. And everywhere there is such an or.
It's probably obvious for the experts, but extremely confusing for everyone else.